### DISCUSSION TITLE
New Source: Kustomer

### DISCUSSION THREAD

**Discussion by rparrapy** [airbytehq/airbyte#35259] [New Connector Request]
## Tell us about the new connector you’d like to have
I'd like to have a new connector to pull data from Kustomer to Snowflake

## Describe the context around this new connector
This is needed to do customer reporting for our marketing team.
The most important endpoints for us are:

- Customers
- Companies
- Customer events
- Conversations
- Notes

Less important but nice to haves would be:
- Teams, Tags, Shortcuts, Users

We would run this sync at least on a daily basis, maybe more frequently.

## Describe the alternative you are considering or using
https://github.com/singer-io/tap-kustomer

**Comment by cheyura** [comment]
**Webhook-based? (no/partially/yes)**
Yes
Similar to form hooks, webhooks are also unique URLs that accept data in an HTTP POST Request. However, webhooks require authentication using an API Key that has the org.hooks role. Currently, webhooks accept JSON and Form-Data content. Webhooks are best suited for server-to-server communication where verifying the authenticity of the sender is important. In Kustomer, posting to a webhook requires a Kustomer API key with org.hook permission. For more information on creating and using API keys, see API Keys/Authentication.

**Available authentication modes (API key/Oauth/other)**
You can authorize externally hosted apps to access the Kustomer API either with a Kustomer API key or with the Kustomer OAuth provider. The Kustomer OAuth provider generates an access token that you can use like a Kustomer API key to authorize API requests.
The mote easy way to send requests to Kustomer is API keys
https://support.kustomer.com/api-keys-SJs5YTIWX
**Creating an account**
Paid 
https://www.kustomer.com/pricing/

**Has Sandbox environment ?**
If you'd like to create a private app to install for your organization, develop and test apps using a sandbox account before you register the app to your Kustomer organization. You can contact sales@kustomer.com to request a new sandbox account or to request a test organization for development.

To create and register apps, you must have at least Administrator access to your Kustomer organization.

**How to populate the account with data?**


**Available streams for sync**

Attachments
Brands
Business Schedules
Companies
Companies (Bulk)
Conversations
Conversations (Bulk)
Conversations (Messages)
Conversations (Tags)
Conversations (Forwards)
Conversations (Time)
Notes (Conversations)
Conversations (Notes)
Customers (Bulk)
Customers (Conversations)
Customers (Drafts)
Customers (Events)
Customers (KObjects)
Customers (Merges)
Customers (Messages)
Customers (Notes)
Customers
Custom Attributes
Drafts (Attachments)
Drafts
KObjects (Custom Objects) (Bulk)
KObjects (Custom Objects)
Klasses
Media
Messages (Bulk)
Messages
Notes (Bulk)
Notes
Tracking Events
Search
SLAs
Spam


**Integration supports incremental sync**
No
https://developer.kustomer.com/kustomer-api-docs/reference/pagination

**Other information/blockers**
For sandbox account assistance from Kustomer representative is needed

**Comment by sherifnada** [comment]
@cheyura just confirming: did you apply for the partnership application as suggested by Kustomer.io ?

**Comment by cheyura** [comment]
> @cheyura just confirming: did you apply for the partnership application as suggested by Kustomer.io ?

@sherifnada Yes, did it second time to be sure

**Comment by rparrapy** [comment]
Hi, I understand that the partnership application is probably a requirement to get connector merged. Any news on that end?

**Comment by cheyura** [comment]
Hi,applied for partnership but no reply
Пт, 30 июля 2021 г. в 16:57, Rodrigo Parra ***@***.***>:

**Comment by sherifnada** [comment]
@rparrapy we can get it merged without a partnership given that we have proof of acceptance tests passing on the PR. I've also reached out to the Kustomer team again. Will follow up on the PR.

**Comment by misteryeo** [comment]
Hey @ycherniaiev just checking, is this issue blocked because of Kustomer? If so, can you update the status of this issue?

**Comment by ycherniaiev** [comment]
Hey @misteryeo we have sandbox but regarding oAuth app is an another story. Private app is available, but for public app we need a partnership with Kustomer

**Comment by misteryeo** [comment]
Thanks @ycherniaiev - marking this as blocked until we can establish that partnership. Thanks!

[a comment by an automated account is not shown]
